packages/dds/sequence/src/test/revertibles.spec.ts

```diff
@@ -90,11 +101,93 @@ describe("Sequence.Revertibles with Local Edits", () => {
+	});
+	it("reverts multiple interval removes", () => {
+		collection.on("deleteInterval", (interval, local, op) => {
+			appendLocalDeleteToRevertibles(sharedString, interval, revertibles);
+		});
+
+		sharedString.insertText(0, "hello world");
+		// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+		const id = collection.add(0, 5, IntervalType.Simple).getIntervalId()!;
+		// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+		const id2 = collection.add(5, 7, IntervalType.Simple).getIntervalId()!;
+		collection.removeIntervalById(id);
+		collection.removeIntervalById(id2);
+
+		revertIntervalRevertibles(sharedString, revertibles.splice(0));
+		assertIntervals(sharedString, collection, [
+			{ start: 0, end: 5 },
+			{ start: 5, end: 7 },
+		]);
+	});
+	it("performs multiple reverts on the same interval", () => {
+		collection.on("addInterval", (interval, local, op) => {
+			appendLocalAddToRevertibles(interval, revertibles);
+		});
+
+		collection.on("deleteInterval", (interval, local, op) => {
+			appendLocalDeleteToRevertibles(sharedString, interval, revertibles);
+		});
+
+		collection.on("changeInterval", (interval, previousInterval, local, op) => {
+			appendLocalChangeToRevertibles(sharedString, interval, previousInterval, revertibles);
+		});
+
+		sharedString.insertText(0, "hello world");
+		// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+		const id = collection.add(0, 5, IntervalType.Simple).getIntervalId()!;
+		collection.change(id, 3, 8);
+		collection.removeIntervalById(id);
+
+		revertIntervalRevertibles(sharedString, revertibles.splice(0));
+		assertIntervals(sharedString, collection, []);
+	});
+	it("performs two local changes, then reverts the first", () => {
+		collection.on("addInterval", (interval, local, op) => {
+			appendLocalAddToRevertibles(interval, revertibles);
+		});
+
+		sharedString.insertText(0, "hello world");
+		// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+		const id = collection.add(0, 5, IntervalType.Simple).getIntervalId()!;
+		collection.change(id, 3, 8);
+
+		revertIntervalRevertibles(sharedString, revertibles.splice(0));
+		assertIntervals(sharedString, collection, []);
+	});
+	it("performs two local changes, acks the second, and then reverts the first", () => {
+		collection.on("addInterval", (interval, local, op) => {
+			appendLocalAddToRevertibles(interval, revertibles);
+		});
+
+		sharedString.insertText(0, "hello world");
+		// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+		const id = collection.add(0, 5, IntervalType.Simple).getIntervalId()!;
+
+		collection.change(id, 3, 8);
+		containerRuntimeFactory.processOneMessage();
+
+		revertIntervalRevertibles(sharedString, revertibles.splice(0));
+		assertIntervals(sharedString, collection, []);
+	});
 });
-describe("Sequence.Revertibles with Remote Edits", () => {
+describe.only("Sequence.Revertibles with Remote Edits", () => {
```

Comment:
I think your cases so far have avoided hitting this, but in the case where the local and remote edits are both of the same kind that you're subscribing to the events on, you'll have to filter to only add to revertibles for local edits.